pkg/aks/client.go

```diff
@@ -133,5 +140,31 @@ func GetSecrets(secretsCache wranglerv1.SecretCache, spec *aksv1.AKSClusterConfi
 	cred.AuthBaseURL = spec.AuthBaseURL
 	cred.BaseURL = spec.BaseURL
 
+	if cred.TenantID == "" {
+		annotations := secret.GetAnnotations()
+		tenantAnno, timestamp := annotations[tenantIDAnnotation], annotations[tenantIDTimestampAnnotation]
+		if tenantAnno != "" && timestamp != "" {
+			parsedTime, err := time.Parse(time.RFC3339, timestamp)
+			if err != nil {
+				return nil, err
+			}
+			if parsedTime.Add(tenantIDTimeout).After(time.Now().UTC()) {
+				cred.TenantID = tenantAnno
+			}
+		}
+		if cred.TenantID == "" {
+			cred.TenantID, err = azureutil.FindTenantID(context.Background(), azure.PublicCloud, string(cred.SubscriptionID))
+			if err != nil {
+				return nil, err
+			}
+			secret.Annotations[tenantIDAnnotation] = cred.TenantID
+			secret.Annotations[tenantIDTimestampAnnotation] = time.Now().UTC().Format(time.RFC3339)
+			// Ignore errors when updating the secret object. If the secret cannot be updated
+			// (perhaps due to a conflict error), the tenant ID will be re-fetched on the next reconcile loop.
+			_, err := secretClient.Update(secret)
```

Comment:
The error should be nil checked and only printed out if it is not nil.

Reply:
fixed